src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -889,21 +898,15 @@ def get_nodepool_labels(self) -> Union[Dict[str, str], None]:
         :return: dictionary or None
         """
         # read the original value passed by the command
-        raw_value = self.raw_param.get("nodepool_labels")
-        # try to read the property value corresponding to the parameter from the `mc` object
-        value_obtained_from_mc = None
-        if self.mc and self.mc.agent_pool_profiles:
-            agent_pool_profile = safe_list_get(
-                self.mc.agent_pool_profiles, 0, None
-            )
-            if agent_pool_profile:
-                value_obtained_from_mc = agent_pool_profile.node_labels
-
-        # set default value
-        if value_obtained_from_mc is not None:
-            nodepool_labels = value_obtained_from_mc
-        else:
-            nodepool_labels = raw_value
+        nodepool_labels = self.raw_param.get("nodepool_labels")
+        # In create mode, try to read the property value corresponding to the parameter from the `mc` object.
+        if self.decorator_mode == DecoratorMode.CREATE:
```

Comment:
Does this means we only ready labels in create mode? Will this function obtain the value of nodepool labels during update mc? Labels are mutable now

Reply:
In the update mode, the parameter value is not obtained from `mc` because we just need to use the value provided by the user to overwrite the value in `mc`.

But in the create mode, `mc` is completely blank, and we want to assemble the user's input into a complete `mc`. Considering that our parameters have a certain dependency or mutual exclusion relationship, or the parameters will be normalized before putting into `mc`, there are even some parameters that do not need user to provide explicitly, but can be dynamically completed by cli.

In order to ensure the consistency of the state and make the value of the parameter more clearly determined, we created the following principle in the create mode, that is, if a parameter has been placed in the `mc` object, then when trying to get the parameter value again, (e.g. cross-validation with other parameters) the value in `mc` shall prevail.